tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Result.java

```diff
@@ -0,0 +1,186 @@
+=======================================================================
+*/
+package org.tensorflow;
+
+import org.tensorflow.proto.framework.RunMetadata;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Logger;
+
+/**
+ * An {@link AutoCloseable} wrapper around a {@link Map} containing {@link Tensor}s.
+ *
+ * <p>When this is closed it closes all the {@link Tensor}s inside it. If you maintain a
+ * reference to a value after this object has been closed it will throw an {@link
+ * IllegalStateException} upon access.
+ */
+public final class Result implements AutoCloseable, Iterable<Map.Entry<String, Tensor>> {
+    @Override
+    public void close() {
+        if (!closed) {
+            closed = true;
+            for (Tensor t : map.values()) {
+                t.close();
+            }
+        } else {
+            logger.warning("Closing an already closed Result");
+        }
+    }
+
+    @Override
+    public Iterator<Map.Entry<String, Tensor>> iterator() {
+        if (!closed) {
+            return map.entrySet().iterator();
+        } else {
+            throw new IllegalStateException("Result is closed");
+        }
+    }
+
+    /**
+     * Returns the number of outputs in this Result.
+     *
+     * @return The number of outputs.
+     */
+    public int size() {
+        return map.size();
+    }
+
+    /**
+     * Gets the set containing all the tensor names.
+     * @return The tensor names set.
+     */
+    public Set<String> keySet() {
+        return Collections.unmodifiableSet(map.keySet());
+    }
+
+    /**
+     * Does this result object have a tensor for the supplied key?
+     * @param key The key to check.
+     * @return True if this result object has a tensor for this key.
+     */
+    public boolean containsKey(String key) {
+        return map.containsKey(key);
+    }
+
+    /**
+     * Gets the value from the container at the specified index.
+     *
+     * <p>Throws {@link IllegalStateException} if the container has been closed, and {@link
+     * IndexOutOfBoundsException} if the index is invalid.
+     *
+     * @param index The index to lookup.
+     * @return The value at the index.
+     */
+    public Tensor get(int index) {
+        if (!closed) {
+            return list.get(index);
+        } else {
+            throw new IllegalStateException("Result is closed");
+        }
+    }
+
+    /**
+     * Gets the value from the container assuming it's not been closed.
+     *
+     * <p>Throws {@link IllegalStateException} if the container has been closed.
+     *
+     * @param key The key to lookup.
+     * @return Optional.of the value if it exists.
+     */
+    public Optional<Tensor> get(String key) {
+        if (!closed) {
+            Tensor value = map.get(key);
+            if (value != null) {
```

Comment:
why not using `Optional.ofNullable` here?

Reply:
Because I always forget that method exists, I'll fix it.